backend/routes.js

```diff
@@ -162,30 +162,50 @@ route.POST('/file', {
       console.error('failAction error:', err)
       return err
     },
-    maxBytes: 6 * MEGABTYE, // TODO: make this a configurable setting
+    maxBytes: 6 * MEGABYTE, // TODO: make this a configurable setting
     timeout: 10 * SECOND // TODO: make this a configurable setting
   }
 }, async function (request, h) {
   try {
     console.log('FILE UPLOAD!')
-    console.log(request.payload)
     const { hash, data } = request.payload
     if (!hash) return Boom.badRequest('missing hash')
     if (!data) return Boom.badRequest('missing data')
-    // console.log('typeof data:', typeof data)
     const ourHash = blake32Hash(data)
     if (ourHash !== hash) {
       console.error(`hash(${hash}) != ourHash(${ourHash})`)
       return Boom.badRequest('bad hash!')
     }
-    await sbp('backend/db/writeFileOnce', hash, data)
+    await sbp('chelonia/db/set', `blob:${hash}`, data)
     return '/file/' + hash
   } catch (err) {
     return logger(err)
   }
 })
 
+// Serve data from Chelonia DB.
+// Note that a `Last-Modified` header isn't included in the response.
 route.GET('/file/{hash}', {
+  cache: {
+    // Do not set other cache options here, to make sure the 'otherwise' option
+    // will be used so that the 'immutable' directive gets included.
+    otherwise: 'public,max-age=31536000,immutable'
+  }
+}, async function (request, h) {
+  const { hash } = request.params
+  console.debug(`GET /file/${hash}`)
+  const blob = await sbp('chelonia/db/get', `blob:${hash}`)
+  if (!blob) {
+    return Boom.notFound()
+  }
+  return h.response(blob).etag(hash)
+})
+
+// Used to serve contract source files and contract manifests.
+// These files are stored in the file system rather than the Chelonia db,
+// therefore `GET /file` could not be used since a different handler was needed.
+// This route is very similar to `GET /assets`, but for now a separate route must be used.
+route.GET('/contract/{hash}', {
```

Comment:
I don't understand what's going on here and the comment isn't enough to help me understand.

> // These files are stored in the file system rather than the Chelonia db,

This doesn't seem to be true?  `POST /file` uses:

```js
await sbp('chelonia/db/set', `blob:${hash}`, data)
```

That is the Chelonia DB, which can be either the filesystem, memory, or sqlite.

The source files should be stored wherever `POST /file` stores them, and that should be equivalent to the above 3 possible backends. Am I missing something?

Reply:
These files are only stored at build time using `chel deploy` or `deployAll`, and these commands will put them in the file system. Yea `POST /file` uses Chelonia DB, `GET /file `does too, that's why a new route endpoint was needed

Reply:
OOHHHHHH!!!

OK, that's an issue... ideally `chel deploy`/`chel deployAll` would deploy it to the database, whatever that happens to be. In this case we're mixing and matching databases (filesystem and sqlite), and that's the problem. `chel deploy` should be instructed which one to use, then this wouldn't be a problem.

We could use `/contract` temporarily, but only as a hack until we fix `chel deploy` to take an argument to accept which type of database we're deploying to